content/src/debug.js

```diff
@@ -201,38 +364,205 @@ function reportSeeeval(method, debugId, length, coordId, elem, args){
   debugRecordsByDebugId[currentDebugId] = record;
 }
 
+function reportEnter(method, debugId, length, coordId, elem, args){
+  stuckComplexity.calls += 1;
+  var recordD = debugRecordsByDebugId[debugId];
+  var recordL = debugRecordsByLineNo[recordD.line];
+  recordD.animated = false;
+  recordL.animated = false;
+}
+
+
 function reportAppear(method, debugId, length, coordId, elem, args){
-  stuckComplexity.moves += 1;
+if (!programChanged) { 
   var recordD = debugRecordsByDebugId[debugId];
-  if (recordD) {
-    if (!recordD.seeeval) {
-      var recordL = debugRecordsByLineNo[recordD.line];
-      recordD.method = method;
-      recordL.method = method;
-      recordD.args = args;
-      recordL.args = args;
+  var recordL = debugRecordsByLineNo[recordD.line];
+  recordD.method = method;
+  recordL.method = method;
+  recordD.args = args;
+  recordL.args = args;
+
+  var currentRecord = debugRecordsByDebugId[currentRecordID];
+  var currentIndex = currentRecord.eventIndex;
+  var currentLocation = traceEvents[currentIndex].location;
+  var currentLine = currentLocation.first_line;
+
+  var prevLine = 0;
+
+  stuckComplexity.moves += 1;
+
+  if (recordD) { 
+    if (!recordD.seeeval){ 
       var index = recordD.eventIndex;
-      var location = traceEvents[index].location.first_line;
+      var line = traceEvents[index].location.first_line;
+      var appear_location = traceEvents[index].location;
+      var tracedIndex = -1; 
+
+      //trace lines that are not animation.
```

Comment:
This is a nice comment.  Add a space after `//`.

Also add a similar comment or two below describing what the other parts of this function are doing.